indigo/indigo/src/main/scala/indigo/shared/events/GlobalEvent.scala

```diff
@@ -88,6 +88,22 @@ case object FullScreenExited extends ViewEvent
   */
 case object FullScreenExitError extends ViewEvent
 
+/** The application has received focus
+  */
+case object ApplicationGainedFocus extends GlobalEvent
+
+/** The game canvas has received focus
+  */
+case object CanvasGainedFocus extends GlobalEvent
+
+/** The application has lost focus
+  */
+case object ApplicationLostFocus extends GlobalEvent
+
+/** The game canvas has lost focus
+  */
+case object CanvasLostFocus extends GlobalEvent
```

Comment:
Please don't forget to add aliases to the package.scala file to help people import these:

https://github.com/PurpleKingdomGames/indigo/blob/c4d33922194bf88eeb07bddb772b1b71d5e848c8/indigo/indigo/src/main/scala/indigo/package.scala#L403